### ISSUE TITLE
Namedtuples checked as tuples when instantiated inline

### ISSUE THREAD

**Issue by mplanchard** [python/mypy#3980]
Thanks for the great library! I use it daily.

I have read through all of the issues on namedtuples, and while I don't think I have seen this issue, I apologize if I missed it or it is obviously caused by an already-reported issue.

When instantiating a namedtuple inline with its definition, the namedtuple is checked as a tuple, leading to incorrect errors.

For example:

```python
from collections import namedtuple

inst_0 = namedtuple('NT0', ['a', 'b'])('a', 'b')  
# error: Too many arguments for "tuple"

inst_1 = namedtuple('NT1', ['a', 'b'])(a='a', b='b')
# error: Unexpected keyword argument "a" for "tuple"
# error: Unexpected keyword argument "b" for "tuple"
# error: Need type annotation for variable
```

Only instantiating with keyword arguments yields the "Need type annotation" error.

The behavior does not change if `import collections` is used rather than `from collections import namedtuple`, or if `inst_1` is defined before `inst_0`.

**Comment by gvanrossum** [comment]
Have you tried `from typing import NamedTuple` instead?

**Comment by mplanchard** [comment]
Using `typing.NamedTuple` doesn't generate the above error (and is functional):

```python
from typing import NamedTuple

inst_0 = NamedTuple('NT0', a=str, b=str)('a', 'b')
inst_1 = NamedTuple('NT0', a=str, b=str)(a='a', b='b')
```

However, calling `NamedTuple` in this way generates a mypy error `error: "NamedTuple" not callable`. 

Going forward, is `typing.NamedTuple` the recommended way to instantiate `namedtuple`s, rather than `collections.namedtuple`?

**Comment by gvanrossum** [comment]
Yes, if you want your types checked you should use `NamedTuple`. The docs show that you have to call it slightly differently. http://mypy.readthedocs.io/en/latest/kinds_of_types.html#named-tuples

**Comment by mplanchard** [comment]
Okay, I will stick to that for now for ensuring that types are checked. 

Sorry if it's a bother, but I am still curious about the `collections.namedtuple()` constructor's return value being correctly interpreted as a namedtuple, unless the return is used within the same statement. 

Maybe it points to my using namedtuples for something they're not designed for, but it seems strange that this does not generate an error:
```python
NT = namedtuple('NT', ['a', 'b'])
nt = NT('a', 'b')
```
while this does:
```python
nt = namedtuple('NT', ['a', 'b'])('a', 'b')
```

I'd be happy to dig through the code to try to figure it out, but I'm not quite sure where to start.

**Comment by gvanrossum** [comment]
That's because namedtuple and NamedTuple are special-cased during the semantic analysis, and only in certain contexts. Since what you're doing is pretty atypical, I don't think it's worth addressing.
